tensorflow/python/keras/benchmarks/keras_examples_benchmarks/README.md

```diff
@@ -0,0 +1,183 @@
+
+## Installing Bazel
+
+This step can be skipped if Bazel is already installed. </br>
+
+We need to use [Bazel](https://bazel.build/) to build targets based on BUILD files. It will take a while for the first time because it will compile all dependencies from your BUILD file. For the next time, Bazel will use the cache and it’ll be much faster. Since we use Ubuntu OS, we can install bazel by using apt repository.
+
+1. Add bazel as package source
+
+   ```shell
+   sudo apt install curl gnupg
+   ```
+
+   ```shell
+   curl https://bazel.build/bazel-release.pub.gpg | sudo apt-key add -
+   ```
+
+   ```shell
+   echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
+   ```
+
+   Before we install the bazel, We should take a look for a bazel version that can build the specific tensorflow version, you can check it from [here](https://www.tensorflow.org/install/source#tested_build_configurations). In addition, you can follow the instructions from [Bazel website](https://docs.bazel.build/versions/3.4.0/install.html).
+
+2. Install Bazel
+
+   ```shell
+   sudo apt update && sudo apt install bazel-`version`
+   ```
+
+## How to run benchmarks
+
+To run benchmarks in [keras/benchmarks](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/keras/benchmarks), please take the following steps:
+
+1. Pull the latest tensorflow repo from github.
+2. Install the Bazel tool which works with tensorflow, please take a look for the Tool installation section.
+3. To run benchmarks with Bazel, use the `--benchmarks=.` flags to specify the benchmarks to run.
+    - To run all benchmarks on CPU
+
+     ```shell
+     bazel run -c opt benchmark_test -- --benchmarks=.
+     ```
+
+    - To run all benchmarks on GPU
+
+     ```shell
+     bazel run run --config=cuda -c opt --copt="-mavx" benchmarks_test -- \ --benchmarks=.
+     ```
+
+    - To run a subset of benchmarks using `--benchmarks` flag, `--benchmarks`: the list of benchmarks to run. The specified value is interpreted as a regular expression and any benchmarks whose name contains a partial match to the regular expression is executed. e.g. `--benchmarks=".*lstm*."`, will run all lstm layer related benchmarks.
+
+## How to add new benchmark tests that use `fit`
+
+To add a new benchmark, please follow the steps:
+
+1. Create your own benchmark test file, `xxxx_benchmark_test.py`.
+2. Import `benchmark_util` to measure and track performance.
+3. Create class which inherits from `tf.test.Benchmark`
+4. Define and load dataset in `__init__` method.
+5. Design and create a model in `_build_model` method.
+6. Define the `benchmark_xxx` method and it will pass essential parameters, which includes `batch_size`, `run_iters`, `train_data` and etc. You can check examples from [here](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/keras/benchmarks/keras_examples_benchmarks).
```

Comment:
Can you explain "it will pass essential parameters" ? Maybe to rephrase it a bit?

"essential parameters" --> "benchmark related hyper parameters".

Reply:
Sounds great to me.